ross/rotor_assembly.py

```diff
@@ -519,31 +519,43 @@ def flatten(l):
         self.df = df
 
     def _check_number_dof(self):
-        number_dof = len(self.shaft_elements[0].dof_mapping()) / 2
+        """Verify the consistency of degrees of freedom.
 
-        for el in self.shaft_elements:
-            if (len(el.dof_mapping()) == 8 and number_dof == 6) or (
-                len(el.dof_mapping()) == 12 and number_dof == 4
-            ):
-                raise Exception(
-                    "The number of degrees o freedom of all elements must be the same! There are SHAFT elements with discrepant DoFs."
-                )
+        This method loops for all the elements, checking if the number of degrees of
+        freedom is consistent.
+        E.g.: inputting 2 shaft elements, one with 4 dof and one with 6, will raise
+        an error.
 
-        for el in self.disk_elements:
-            if (len(el.dof_mapping()) == 4 and number_dof == 6) or (
-                len(el.dof_mapping()) == 6 and number_dof == 4
-            ):
-                raise Exception(
-                    "The number of degrees o freedom of all elements must be the same! There are DISK elements with discrepant DoFs."
-                )
+        Raises
+        ------
+        Exception
+            Error pointing out difference between the number of DoF's from each element
+            type.
 
-        for el in self.bearing_elements:
-            if (len(el.dof_mapping()) == 2 and number_dof == 6) or (
-                len(el.dof_mapping()) == 3 and number_dof == 4
-            ):
-                raise Exception(
-                    "The number of degrees o freedom of all elements must be the same! There are BEARING elements with discrepant DoFs."
-                )
+        Returns
+        -------
+        number_dof : int
+            Number of degrees of freedom from the adopted shaft element.
+        """
+
+        number_dof = len(self.shaft_elements[0].dof_mapping())
```

Comment:
@verg1lio I think you need to divide by 2 in the end of the line
`len(self.shaft_elements[0].dof_mapping())` returns 8 or 12 (related to 2 nodes). And we need the number of DoF's for a single node, in this case. Thus, the other comparations will make sense.

Reply:
Hi @rodrigomoliveira1 , good morning. We realized that and did it on the last commit. It seems you have an outdated version there.